### DISCUSSION TITLE
QRB and QTF calculation in case of long gridsquare (10 symbols)

### DISCUSSION THREAD

**Discussion by R1BLH** [magicbug/Cloudlog#2811] [General]
If the gridsquare is set exactly (10 characters, e.g. FN41iv01aq), the calculation is incorrect. 
If the gridsquare is 8 characters long, the calculation is correct.
See screenshots. 

QSO:
![2023-12-12_03-28-28](https://github.com/magicbug/Cloudlog/assets/16667173/87171c95-3e2b-4285-a60e-8fdbd5400af4)

QRB and QTF calculation:
![2023-12-12_03-29-01](https://github.com/magicbug/Cloudlog/assets/16667173/97ea0825-b3bf-408a-9790-f98df1160481)

QRB and QTF calculation for manually shortened gridsquare (8 symbols):
![2023-12-12_03-30-07](https://github.com/magicbug/Cloudlog/assets/16667173/96d52f5f-6429-4994-992f-03f0cdbbf75e)

**Comment by HB9HIL** [comment]
Maidenhead Locators with 10 characters are not supported in Cloudlog. And it's not planned to support this. I'm sorry.
Cloudlog supports up to 8 characters.

The normal maidenhead locator has 4-6 characters. Some have 8. 
10 Characters is something special and the most services do not support more then 6 characters.

**Reply by HB9HIL** [reply]
perhaps I was bit fast with that 😄

**Comment by magicbug** [comment]
Yep isn't supported that said it /libriaries/qra.php is the only file that would need to be adjusted to support this.

**Reply by R1BLH** [reply]
Is it a good idea to automatically use only the first 8 symbols of a 10 symbol locator when calculating for shortwave QSOs?

**Reply by HB9HIL** [reply]
@cats-shadow For Shortwave it's absolute okay to use only 6 character locators, For example `FN41IV`

**Reply by HB9HIL** [reply]
PS: On shortwave it makes not such a big difference between 6 symbol locators.

**Reply by R1BLH** [reply]
@HB9HIL yes, i know. :)

**Comment by phl0** [comment]
I am gonna have a look :)

**Reply by magicbug** [reply]
Would be good would please the microwavers etc

**Comment by phl0** [comment]
I made a fix in https://github.com/magicbug/Cloudlog/pull/2821. Please test and report.

**Comment by phl0** [comment]
This has been merged to dev. Please close here if issue is gone.

**Reply by R1BLH** [reply]
@phl0 Now the locator is automatically truncated to 6 symbols?

![изображение](https://github.com/magicbug/Cloudlog/assets/16667173/e7a831f4-4512-4f02-89b7-fac4e9ec37ba)

**Reply by phl0** [reply]
No it is not. Did you re-enter that QSO or just displayed?

**Reply by R1BLH** [reply]
This is the same QSO.
This is really strange. The information in the database is changed! I cannot remember if I did it myself.
By the way, I found another QSO with 10 symbol locator. Everything is OK now.
![изображение](https://github.com/magicbug/Cloudlog/assets/16667173/c3a6f0e5-e069-4f75-a5dd-bd51678a04fe)

Interestingly, the coordinate values in this case have only three digits in the fractional part, although the locator indicates a more precise position. 
So the accuracy of the calculation is determined by the minimum accuracy of one of the two locators?
![изображение](https://github.com/magicbug/Cloudlog/assets/16667173/3515679d-6a18-4303-9d04-00798489d461)

I changed my locator to a more precise one (10 characters), but nothing changed (except distance). The coordinates are still displayed with three decimal places.
![изображение](https://github.com/magicbug/Cloudlog/assets/16667173/47b6183f-7f93-48ad-ad12-32eef656fbff)

**Reply by phl0** [reply]
Oh well I did not check the rounding of the results. Lets see if we can improve that.
